site/source/tutorials/007_theming/index.md

```diff
@@ -0,0 +1,136 @@
+
+{% instruction 'Theme the worker widget' %}
+
+In order to theme the `worker` widget, we will need to create a `worker.m.css` file within our theme directory and and use it within our `theme.ts` file. The naming here is important as the object key of the exported theme must match the naming of the widget's css file.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/theme.ts' lines:3,9,11,16 %}
+
+Lets start by creating a red `worker` so that we can see our theme has been applied correctly.
+
+```css
+/* worker.m.css */
+.root {
+	background: red;
+}
+```
+
+In order to apply our theme, we will need to use a `themeInjector`. Alternatively we *could* set `theme` on the top level widget and pass it down via properties to every child widget but this approach is not very elegant. We can create a `themeInjector` using the `registerThemeInjector` function exported by `ThemeableMixin`. So let's go ahead and update our `main.ts` file to import our theme and create a `themeInjector`.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' %}
+
+If you open up your web browser now you should see that the `worker` backgrounds are now `red`.
+
+{% instruction 'Use variables and complete the worker theme' %}
+
+The Dojo 2 build system allows us new `css` features such as `css-custom-properties` by using `postCss` to process our `.m.css` files. We can use these to add variables to `worker.m.css` and to complete it's theme.
+Let go ahead and create `themes/dojo/variables.css` (notice that this file does not have a `.m.css` extension as it is not a `css-module` file).
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/variables.css' lang:css %}
+
+We can now use these variables in `worker.m.css` to create our fully themed `worker`.
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/worker.m.css' lang:css %}
+
+{% section %}
+
+## Theming Dojo widgets
+
+So far in this tutorial we have themed our custom `worker` widget, but how do we target dojo widgets with our theme. To demonstrate this, we will theme the `workerForm` widget as it contains both domNodes and dojo widgets to be themed.
+
+{% instruction 'Lets create `workerForm.m.css` and include it in our theme.ts' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/workerForm.m.css' lang:css %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/themes/dojo/theme.ts' lines:3,5,9,11,13,16 %}
+
+This should be familiar to you by now as we covered theming `worker` in the last section. Now to theme the Dojo 2 `TextInput` that `WorkerForm` uses, we will need to create a `dojoTextInput.m.css` file and export it from `theme.ts` using a special theme key which is prefixed with `dojo-`. This prefix helps to ensure that our own widget theme files do not clash with dojo specific widgets and makes it easy to recognise when we are targeting one of the other.
```

Comment:
This should be familiar to you by now as we covered theming `worker` in the last section. -> This should be familiar from theming the `worker` in the previous section.